### ISSUE TITLE
`{deno:{namespace:false}}` worker option doesn't remove `Deno` namespace

### ISSUE THREAD

**Issue by josephrocca** [denoland/deno#26056]
**Version**: Deno `1.46.x` and `2.x`

This logs `Received from worker: Hello from Worker! 2.0.0-rc.10+053894b` instead of throwing a reference error:
```js
const workerBlob = new Blob([`
  self.onmessage = async function(e) {
    self.postMessage('Hello from Worker! ' + Deno.version.deno);
  }
`], { type: 'application/javascript' });
  
const workerURL = URL.createObjectURL(workerBlob);
  
const worker = new Worker(workerURL, {type:"module", deno:{permissions:"none", namespace:false}});
  
worker.onmessage = function(e) {
  console.log('Received from worker:', e.data);
};
worker.postMessage('Hello from main thread!');
```

This issue may be more of a feature request, since it's not explicitly mentioned in the docs that `namespace:false` disables Deno stuff. It only says:

> Starting in v1.22 the Deno namespace is available in worker scope by default. To enable the namespace in earlier versions pass deno: { namespace: true } when creating a new worker.

Note if `delete globalThis.Deno` is equivalent, this may not be an ideal solution for me, because part of the reason I want to remove the Deno stuff is to reduce the memory overhead of the worker.

**Comment by lucacasonato** [comment]
Yeah, this is not a supported feature. You should run `delete globalThis.Deno`. Memory overhead of the worker would not be different if we implemented this using `namespace: false`.

**Comment by josephrocca** [comment]
> Yeah, this is not a supported feature.

Ah okay, no worries. I think it should be supported, and since satyarohith has labelled this issue as a suggestion, I'll leave it open, but of course feel free to close this if there's no intention/desire to support `false` as a valid value, since the workaround it easy anyway.

**Comment by josephrocca** [comment]
One caveat, IIUC, to a simple `delete globalThis.Deno` at the top of a Worker script is that if the ~untrusted/userland code has static imports, then those imports can access the Deno namespace because the static imports are hoisted and hence executed before `delete globalThis.Deno`. So I think `delete globalThis.Deno` would need to be put inside a static import, which is then included before the untrusted code.

**Comment by 0f-0b** [comment]
Since you are running untrusted code inside workers, I would like to point out that it is very easy for untrusted code to crash the whole process even if it only has access to JavaScript builtins.

**Comment by josephrocca** [comment]
@0f-0b Indeed, I came to that conclusion after this investigation:

* https://github.com/denoland/deno/issues/26202

Currently I've resigned myself to use a single process per untrusted code - which is unfortunate because it costs a lot more RAM. I still run code within a worker - my idea being that this way I can allow multiple instances of that code within the same process (i.e. one worker per instance). If an author writes bad code, the most they can do is crash that instance and all of *their* other instances.

But thank you for taking the time to mention this, it's kind of you. (And I'd be glad to hear any other thoughts you have, since I'm early enough in the development still that there's room for change. My use case is long-running servers, btw, not serverless type scenario like CF Workers or Deno Deploy.)

**Comment by Jobians** [comment]
Any updates?
